packages/flutter_tools/test/general.shard/channel_test.dart

```diff
@@ -36,6 +36,17 @@ void main() {
       // and when run locally the list of branches might be different
       // so we check for the header text rather than any specific channel name.
       expect(testLogger.statusText, contains('Flutter channels:'));
+
+      // if any branches are available
+      // check if order of available official branches are in order of stability
+      int prev = -1;
+      for (final String branch in FlutterVersion.officialChannels) {
```

Comment:
Does this work with something like `expect(testLogger.statusText.split('\n'), containsInOrder(FlutterVersion.officialChannels))`?

Reply:
There's a similar matcher called containsAllInOrder, but it can't be used in this situation as a particular official branch might be missing.

Reply:
It wont, because the output from git should be mocked in the test. Also, instead of placing this expect in the setup, move to its own test case

Reply:
Done